internal/store/device.go

```diff
@@ -181,88 +185,105 @@ func (s *DeviceStore) Get(ctx context.Context, orgId uuid.UUID, name string) (*a
 	return &apiDevice, nil
 }
 
-func (s *DeviceStore) CreateOrUpdate(ctx context.Context, orgId uuid.UUID, resource *api.Device, fieldsToUnset []string, fromAPI bool, callback DeviceStoreCallback) (*api.Device, bool, error) {
+func (s *DeviceStore) getExistingRecord(name string, orgId uuid.UUID) (*model.Device, bool, error) {
```

Comment:
Nice helper function.  Let's replace the logic in Get() with this?

Reply:
The semantics is a bit different.  The `getExistingRecord` returns nil error in case the record does not exist.